src/parser/statement.js

```diff
@@ -303,31 +304,37 @@ pp.parseForStatement = function (node) {
     return this.parseFor(node, init);
   }
 
+  let usedNowKeyword = false;
+  if (this.hasPlugin("lightscript") && (!forAwait)) {
+    // idx, elem, key, or val begins a LS-enhanced loop.
+    if (
+      (this.isContextual("idx") || this.isContextual("elem") || this.isContextual("key") || this.isContextual("val"))
+    ) {
+      return this.parseEnhancedForIn(node);
+    }
+
+    // Possibly for now...in
+    if (this.match(tt._now)) {
```

Comment:
.. ah, sorry, nope – would break for `for (;;) { print('infinite!') }` loops. 

But, this can be `usedNowKeyword = this.eat(tt._now)`